src/rust/engine/graph/src/entry.rs

```diff
@@ -434,7 +444,7 @@ impl<N: Node> Entry<N> {
             entry_id,
             run_token,
             generation,
-            if self.node.cacheable() {
+            if self.cacheable_with_output(Some(result.as_ref())) {
```

Comment:
This should either call both methods, or make a guarantee that `cacheable_with_output` calls `cacheable`: the default implementation does, but not the others.